### PR TITLE
secure deployment considerations guide

### PR DESCRIPTION
**Pull request by nnshah1**
Adding initial guide highlighting considerations for deploying Triton Inference Server securely

### CODE REVIEW

File: docs/customization_guide/deploy.md [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+
+# Secure Deployment Considerations
+
+The Triton Inference Server project is designed for flexibility and
+allows developers to create and deploy inferencing solutions in a
+variety of ways. Developers can deploy Triton as an http server, a
+grpc server, a server supporting both, or embed a Triton server into
+their own application. Developers can deploy Triton locally or in the
+cloud, within a kubernetes cluster, behind an API gateway or as a
+standalone process.  This guide is intended to provide some key points
```
Comment (nv-kmcgill53) [inline docs/customization_guide/deploy.md:37]:
```suggestion
their own application. Developers can deploy Triton locally, in the
cloud within a kubernetes cluster behind an API gateway, or as a
standalone process.  This guide is intended to provide some key points
```

File: docs/customization_guide/deploy.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+> [!IMPORTANT]
+> Ultimately the security of a solution based on Triton
+> is the responsibility of the developer building and deploying that
+> solution. When deploying in production settings please have security
+> experts review any potential risks and threats.
+
+> [!WARNING]
+> Allowing dynamic updates to the model repository can lead
+> to arbitrary code execution. Model repository access control is
+> critical in production deployments. Unless required for operation, recommended
```
Comment (nv-kmcgill53) [inline docs/customization_guide/deploy.md:52]:
```suggestion
> critical in production deployments. Unless required for operation, it's recommended
```

File: docs/customization_guide/deploy.md [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+initialization is complete and server is healthy.
+
+##### `--model-control-mode <string>, default "none"`
+
+Specifies the mode for model management.
+
+> [!WARNING]
+> Allowing dynamic updates to the model repository can lead
+> to arbitrary code execution. Model repository access control is
+> critical in production deployments. Unless required for operation, recommended
```
Comment (nv-kmcgill53) [inline docs/customization_guide/deploy.md:168]:
```suggestion
> critical in production deployments. Unless required for operation, it's recommended
```

**Review by nv-kmcgill53 — CHANGES_REQUESTED** [review]
Just a couple of grammatical edits, other than that LGTM.

**Review by dzier — DISMISSED** [review]

File: docs/customization_guide/deploy.md (hunk 1 above)
Comment (nnshah1) [reply docs/customization_guide/deploy.md:37]: on this point - my goal in the list is to imply the flexibility of options someone deploying Triton has at their disposal - so I'd like to keep them as distinct - however maybe there is a better phrasing or list of options that would resonate? I'd like to keep "locally or in the cloud" as one set of hypothetical targets... maybe the next option should be 'within a kubernetes cluster or as a standalone service, behind an API gateway or in an isolated environment. ...

**Review by Tabrizian — COMMENTED** [review]
Is it possible to add a link to this guide in the table of contents?

**Review by Tabrizian — APPROVED** [review]

**Review by nv-kmcgill53 — APPROVED** [review]

**Pull request merged**